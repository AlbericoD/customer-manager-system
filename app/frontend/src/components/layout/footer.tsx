import React, { FunctionComponent } from 'react';
import { Segment, List, Container } from 'semantic-ui-react';

const { Item } = List;
export interface IFooter {
  href: string;
  text: string;
}
interface IProps {
  links: IFooter[];
}
export const Footer: FunctionComponent<IProps> = ({ links }): JSX.Element => (
  <Segment inverted vertical style={{ padding: '0.7em 0em', flex: 'none' }}>
    <Container textAlign='center'>
      <List horizontal inverted divided link size='small'>
        {links.map((item, index) => (
          <Item key={index} as='a' href={item.href} target='__blank'>
            {item.text}
          </Item>
        ))}
      </List>
    </Container>
  </Segment>
);
