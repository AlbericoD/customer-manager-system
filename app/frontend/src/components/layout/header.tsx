import React, { FunctionComponent } from 'react';
import { Menu, Container } from 'semantic-ui-react';

const { Item } = Menu;

export interface IHeader {
  brand: string;
  appName: string;
}
export const Header: FunctionComponent<IHeader> = ({ brand, appName }): JSX.Element => (
  <Menu fixed='top' inverted>
    <Container>
      <Item header>{brand}</Item>
      <Item as='p'>{appName}</Item>
    </Container>
  </Menu>
);
