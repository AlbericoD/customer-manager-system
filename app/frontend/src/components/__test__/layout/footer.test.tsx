import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import ReactDOM from 'react-dom';
import Adapter from 'enzyme-adapter-react-16';
import { Footer } from '../../layout';

Enzyme.configure({
  adapter: new Adapter()
});

describe('Component Footer', () => {
  let props = [{ href: 'test', text: 'test' }];
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Footer links={props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  it('render snapshot', () => {
    const component = shallow(<Footer links={props} />);
    expect(component).toMatchSnapshot();
  });
});
