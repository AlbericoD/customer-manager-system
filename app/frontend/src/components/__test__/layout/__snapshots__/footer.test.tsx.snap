// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Footer render snapshot 1`] = `
<Segment
  inverted={true}
  style={
    Object {
      "flex": "none",
      "padding": "0.7em 0em",
    }
  }
  vertical={true}
>
  <Container
    textAlign="center"
  >
    <List
      divided={true}
      horizontal={true}
      inverted={true}
      link={true}
      size="small"
    >
      <ListItem
        as="a"
        href="test"
        key="0"
        target="__blank"
      >
        test
      </ListItem>
    </List>
  </Container>
</Segment>
`;
